src/clients/openapi.js

```diff
@@ -0,0 +1,111 @@
+import { get } from 'lodash';
+
+import { getConfig, getMetadata } from '@globality/nodule-config';
+import axios from 'axios';
+import OpenAPI from '../client';
+import {
+    buildRequestLogs,
+    logFailure,
+    logSuccess,
+} from './logging';
+
+/* Inject mock and testing adapters.
+ */
+export function buildAdapter(context) {
+    const { name, operationName } = context;
+    const metadata = getMetadata();
+    if (!metadata.testing) {
+        return null;
+    }
+
+    const key = `clients.mock.${name}.${operationName}`;
+    const mock = getConfig(key);
+
+    if (mock) {
+        return mock;
+    }
+
+    return () => {
+        throw new Error(`OpenAPI operation ${name}.${operationName} is not mocked`);
+    };
+}
+
+
+/* Inject standard headers.
+ */
+export function buildHeaders(context, req) {
+    const headers = {
+        'Content-Type': 'application/json; charset=utf-8',
+    };
+
+    // pass the current service's name
```

Comment:
Note that our handing of `X-Request-Foo` is a *framework* decision and probably doesn't belong as something baked into a *library* too deeply. In this case, I'd probably build a new abstraction around header forwarding, keep this implementation as the default behavior _within_ that abstraction, ensure that it can be swapped out (e.g. with a noop), and document it well.